Franklin/Service Layer/Networks/NetworksService.swift

```diff
@@ -47,4 +49,39 @@ public class NetworksService: INetworksService {
         let network = Web3Network(id: id, name: name, endpoint: endpoint)
         return network
     }
+    
+    public func isNetworkExists(network: Web3Network) -> Bool {
+        if network.isXDai()
```

Comment:
Kind of the previous comment - have a distinction on the DB level between custom and built-in networks. Will also require tokens to depend on the network at the DB level, so you can much easier filter relevant ones